### PR TITLE
Tutorials for the docs

### PR DESCRIPTION
**Pull request by kbevers**
Tutorials on using different aspects of PROJ. Originally made for the EUREF meeting in 2019.

### CODE REVIEW

File: docs/source/tutorials/EUREF2019/answers.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                +step +proj=vgridshift +grids=us_nga_egm96_15.tif
+
+        2.1:    In principle, yes - practically, no.
+        2.2:    Which grids is applied first depends on how the systems are defined.
+                If a geoid model is fitted to a certain datum, you need to make sure that
+                the input to the vgridshift operation is in that datum. That is your
+                responsibility to check - PROJ can't figure that out on its own. In this
+                case, the EGM96 geoid is fitted to WGS84. The BETA2007 grid results in
+                ETRS89 coordinates which for all but the most high accuracy application
+                can be regarded as equvialent to WGS84. Hence we do the horizontal grid
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/answers.rst]:
```suggestion
                can be regarded as equivalent to WGS84. Hence we do the horizontal grid
```

File: docs/source/tutorials/EUREF2019/exercises/conversions.gie [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+-------------------------------------------------------------------------------
+
+
+3. Geodetic to cartesian conversion
+-------------------------------------------------------------------------------
+Some transformations, most notably the Helmert transformation, operates on
+cartesian geocentric coordinates. Geodetic coordinates (latitude and longigude)
+is the most commonly used coordinate representation. It is only natural to have
+a way to convert between the two formats.
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/conversions.gie]: maybe
```suggestion
a way to convert between the two representations.
```

File: docs/source/tutorials/EUREF2019/exercises/conversions.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    Remember that the axisswap operation doesn't know anything about the
+    nature of the coordinate that is passed to it - it only cares about the
+    the order of the input.
+
+.. literalinclude:: conversions.gie
+  :lines: 77-81
+
+3. Geodetic to cartesian conversion
+------------------------------------
+Some transformations, most notably the Helmert transformation, operates on
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/conversions.rst]:
```suggestion
Some transformations, most notably the Helmert transformation, operate on
```

File: docs/source/tutorials/EUREF2019/exercises/conversions.gie [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+accept      140.0     75.0    # somewhere in Siberia
+expect       75.0    140.0
+
+-------------------------------------------------------------------------------
+
+
+3. Geodetic to cartesian conversion
+-------------------------------------------------------------------------------
+Some transformations, most notably the Helmert transformation, operates on
+cartesian geocentric coordinates. Geodetic coordinates (latitude and longigude)
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/conversions.gie]:
```suggestion
cartesian geocentric coordinates. Geodetic coordinates (latitude and longitude)
```

File: docs/source/tutorials/EUREF2019/exercises/conversions.rst [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+cartesian geocentric coordinates. Geodetic coordinates (latitude and longigude)
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/conversions.rst]:
```suggestion
cartesian geocentric coordinates. Geodetic coordinates (latitude and longitude)
```

File: docs/source/tutorials/EUREF2019/exercises/conversions.rst [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+is the most commonly used coordinate representation. It is only natural to have
+a way to convert between the two formats.
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/conversions.rst]:
```suggestion
a way to convert between the two representations.
```

File: docs/source/tutorials/EUREF2019/exercises/gridshift.gie [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+Define the operation that converts from DE_DHDN to ETRS89 using a gridshift.
+
+Hints:
+
+    - Consult the documentation to find out how to set up the transformation:
+      https://proj.org/operations/transformations/hgridshift.html
+
+    - Find the relevant grid name by inspecting the output of
+
+        projinfo -s DHDN -t ETRS89 -o WKT2_2018
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/gridshift.gie]:
```suggestion
        projinfo -s DHDN -t ETRS89 -o WKT2_2019
```

File: docs/source/tutorials/EUREF2019/exercises/gridshift.gie [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+      Note that two operations are returned, look for the one which uses the
+      "NTv2" method.
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/gridshift.gie]: as we substitute the original NTv2 grid referenced by EPSG to our geotiff grids, the method name is also changed (might have been a change that has landed after the tutorial was written)
```suggestion
      "HORIZONTAL_SHIFT_GTIFF" method.
```

Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/gridshift.gie]: as we substitute the original NTv2 grid referenced by EPSG to our geotiff grids, the method name is also changed (might have been a change that has landed after the tutorial was written)
```suggestion
      "HORIZONTAL_SHIFT_GTIFF" method.
```

File: docs/source/tutorials/EUREF2019/exercises/gridshift.gie [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+accept      13.0            52.0            0.0
+expect      12.9983317082   51.9986488216   0.0
+
+-------------------------------------------------------------------------------
+
+
+2. Vertical gridshifting
+-------------------------------------------------------------------------------
+The most common use case for vertical grid shifts is conversion from
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/gridshift.gie]: pedantically, shouldn't that be 
```suggestion
The most common use case for vertical grid shifts is transformation from
```

File: docs/source/tutorials/EUREF2019/exercises/gridshift.rst [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+Define the operation that converts from DE_DHDN to ETRS89 using a gridshift.
+
+.. hint:: See :ref:`hgridshift`
+
+.. hint:: 
+    
+    Find the relevant grid name by inspecting the output of
+
+        ``projinfo -s DHDN -t ETRS89 -o WKT2_2018``
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/gridshift.rst]:
```suggestion
        ``projinfo -s DHDN -t ETRS89 -o WKT2_2019``
```

File: docs/source/tutorials/EUREF2019/exercises/gridshift.rst [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+      Note that two operations are returned, look for the one which uses the
+      "NTv2" method.
+
+.. literalinclude:: gridshift.gie
+  :lines: 49-53
+
+2. Vertical gridshifting
+-------------------------
+
+The most common use case for vertical grid shifts is conversion from
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/gridshift.rst]:
```suggestion
The most common use case for vertical grid shifts is transformation from
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.gie [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+exercise we will use a transformation between ED50 in Italy (Sardinia) and
+WGS84.
+
+
+Hints:
+
+    - You can find the transformations parameters by running the following
+      command:
+
+            projinfo -o WKT2_2018 -k operation EPSG:1142
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.gie]:
```suggestion
            projinfo -o WKT2_2019 -k operation EPSG:1142
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.gie [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+using Position Vector. See the PROJ documentation on the Helmert transformation
+for further detail.
+
+
+Hints:
+
+    - You can find the transformations parameters by running the following
+      command:
+
+        projinfo -o WKT2_2018 -k operation EPSG:1626
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.gie]:
```suggestion
        projinfo -o WKT2_2019 -k operation EPSG:1626
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.gie [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+
+accept      3496723.5936    743251.5442  5264442.2361
+expect      3496639.7297    743156.3657  5264324.9341
+
+-------------------------------------------------------------------------------
+
+
+3. Position vector/Coordinate frame
+-------------------------------------------------------------------------------
+Effectively, the difference between the two rotation convetions is the sign of
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.gie]:
```suggestion
Effectively, the difference between the two rotation conventions is the sign of
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.gie [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+the parameters as reported by projinfo (using the coordinate frame convention).
+Adapt the parameters to the position vector convention in 3b.
+
+
+Hints:
+
+    - You can find the transformations parameters by running the following
+      command:
+
+        projinfo -o WKT2_2018 -k operation EPSG:1309
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.gie]:
```suggestion
        projinfo -o WKT2_2019 -k operation EPSG:1309
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.gie [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+between various realizations of ITRS and ETRS89. In this exercise we look at the
+transformation from ITRF2008 to ITRF2014.
+
+
+Hints:
+
+    - You can find the transformations parameters by running the following
+      command:
+
+            projinfo -o WKT2_2018 -s ITRF2008 -t ITRF2014
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.gie]:
```suggestion
            projinfo -o WKT2_2019 -s ITRF2008 -t ITRF2014
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.rst [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+of use it is still commonly used between legacy and modern systems. In this
+exercise we will use a transformation between ED50 in Italy (Sardinia) and
+WGS84.
+
+
+.. hint::
+    You can find the transformations parameters by running the following
+    command::
+
+        projinfo -o WKT2_2018 -k operation EPSG:1142
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.rst]:
```suggestion
        projinfo -o WKT2_2019 -k operation EPSG:1142
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.rst [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+using Position Vector. See the PROJ documentation on the Helmert transformation
+for further detail.
+
+
+.. hint::
+    
+    You can find the transformations parameters by running the following
+    command::
+
+        projinfo -o WKT2_2018 -k operation EPSG:1626
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.rst]:
```suggestion
        projinfo -o WKT2_2019 -k operation EPSG:1626
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.rst [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+First find the parameters for the EPSG:1309 transformation. In 3a below, enter
+the parameters as reported by projinfo (using the coordinate frame convention).
+Adapt the parameters to the position vector convention in 3b.
+
+.. hint::
+
+    You can find the transformations parameters by running the following
+    command::
+
+        projinfo -o WKT2_2018 -k operation EPSG:1309
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.rst]:
```suggestion
        projinfo -o WKT2_2019 -k operation EPSG:1309
```

File: docs/source/tutorials/EUREF2019/exercises/helmert.rst [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+transformations are needed. A good example of such a use case is transformations
+between various realizations of ITRS and ETRS89. In this exercise we look at the
+transformation from ITRF2008 to ITRF2014.
+
+.. hint::
+    
+    You can find the transformations parameters by running the following
+    command::
+
+        projinfo -o WKT2_2018 -s ITRF2008 -t ITRF2014
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/helmert.rst]:
```suggestion
        projinfo -o WKT2_2019 -s ITRF2008 -t ITRF2014
```

File: docs/source/tutorials/EUREF2019/exercises/pipelines.gie [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+As we have seen in the previous exercises, fundamentally, the Helmert
+operation works on cartesian coordinates. So when the input coordinates
+are geodetic, we must convert them to the cartesian space before
+applying the Helmert transformation.
+
+Similarly, we must convert the transformed cartesian coordinates back to
+geodetic space to be able to compare the coordinates before and after
+transformation.
+
+With that in mind, now create a transformation pipeline that take geodetic
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/pipelines.gie]:
```suggestion
With that in mind, now create a transformation pipeline that takes geodetic
```

File: docs/source/tutorials/EUREF2019/exercises/pipelines.gie [hunk 22]
```diff
@@ -0,0 +1,171 @@
...
+back-projection -> geodetic to cartesian conversion -> Helmert transformation ->
+cartesian to geodetic conversion -> re-projection
+
+... a very common recipe for transformations between projected coordinate systems.
+
+Hints:
+
+    - Find the relevant Helmert parameters by running:
+
+        projinfo -k operation EPSG:1626 -o WKT2_2018
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/pipelines.gie]:
```suggestion
        projinfo -k operation EPSG:1626 -o WKT2_2019
```

File: docs/source/tutorials/EUREF2019/exercises/pipelines.rst [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+#. Cartesian to geodetic conversion
+#. Re-projection
+
+which is a very common recipe for transformations between projected coordinate systems.
+
+.. hint:: 
+    
+    Find the relevant Helmert parameters by running::
+
+        projinfo -k operation EPSG:1626 -o WKT2_2018
```
Comment (rouault) [inline docs/source/tutorials/EUREF2019/exercises/pipelines.rst]:
```suggestion
        projinfo -k operation EPSG:1626 -o WKT2_2019
```

**Comment by rhuijben** [comment]
Quoting the build error
```Tabs are bad, please use four spaces in .rst files.```

**Pull request merged**